### PR TITLE
feat: playlist functionality

### PR DESCRIPTION
**Pull request by harshitadaswani**
added playlist functionality using context, reducer having a modal

### CODE REVIEW

File: src/components/playlist/modal/PlaylistModal.jsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                  );
+                })}
+              </div>
+            </form>
+            <div className="dialog-actions border-top p-xs m-xs">
+              <div
+                className="dialog-action txt-cursor"
+                onClick={() => {
+                  setOpenModal(false);
+                }}
```
Comment (JantuDeb) [inline src/components/playlist/modal/PlaylistModal.jsx]: ` onClick={() =>setOpenModal(false)}`

File: src/context/PlaylistContext.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+const PlaylistContext = createContext(defaultPlaylistValue);
+
+export const PlaylistProvider = ({ children }) => {
+  const [openModal, setOpenModal] = useState(false);
+  const [playlistState, playlistDispatch] = useReducer(
+    PlaylistReducer,
+    initialPlaylistValue
+  );
+
+  console.log(playlistState);
```
Comment (JantuDeb) [inline src/context/PlaylistContext.jsx]: remove `log`

**Review by JantuDeb — APPROVED** [review]
Overall the code looks fine, you can add the create and delete functionality for playlist

File: src/components/playlist/modal/PlaylistModal.jsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                          <input
+                            type="checkbox"
+                            name="items"
+                            onChange={(e) => playlistHandler(e, item.id)}
+                            checked={isVideoInPlaylist(
+                              playlistState,
+                              video._id,
+                              item.id
+                            )}
+                          />{" "}
```
Comment (rkapoor10) [inline src/components/playlist/modal/PlaylistModal.jsx:62]: remove this `{" "}`

**Review by rkapoor10 — APPROVED** [review]
- the code is well structured
- liked the use of toast utils

overall looks good to merge

File: src/components/playlist/modal/PlaylistModal.jsx (hunk 3 above)
Comment (harshitadaswani) [reply src/components/playlist/modal/PlaylistModal.jsx:62]: prettier added this, will remove it. Thanks for pointing it out!

File: src/context/PlaylistContext.jsx (hunk 2 above)
Comment (harshitadaswani) [reply src/context/PlaylistContext.jsx]: done

File: src/components/playlist/modal/PlaylistModal.jsx (hunk 1 above)
Comment (harshitadaswani) [reply src/components/playlist/modal/PlaylistModal.jsx]: done

**Pull request merged**